src/System.Net.Sockets/tests/FunctionalTests/SelectTest.cs

```diff
@@ -264,8 +264,7 @@ public static void Select_AcceptNonBlocking_Success()
                 }
 
                 // Give the task 5 seconds to complete; if not, assume it's hung.
-                bool completed = t.Wait(5000);
-                Assert.True(completed);
+                await t.TimeoutAfter(5000);
```

Comment:
Consider modifying the TimeoutAfter extension to include an error message that indicates what the timeout was.